File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def _DFT_inorderHelp(self, currNode, treelist):
        if not currNode:
            return 
        self._DFT_inorderHelp(currNode.left, treelist)
        treelist.append(currNode.val)
        self._DFT_inorderHelp(currNode.right, treelist)

    def kthSmallest(self, root, k):
        treelist = []
        self._DFT_inorderHelp(root, treelist)
        return treelist[k-1]

File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
int __help(struct TreeNode* curr, int* order, int k) {
    if (curr == NULL) return -1;
    int left = __help(curr->left, order, k);
    if (left != -1) return left;
    (*order) ++;
    if (*order == k) return curr->val;
    return __help(curr->right, order, k);
}

int kthSmallest(struct TreeNode* root, int k) {
    int order = 0;
    return __help(root, &order, k);
}